src/ApiService/ApiService/onefuzzlib/orm/Orm.cs

```diff
@@ -150,13 +162,32 @@ public IAsyncEnumerable<T> SearchByRowKeys(IEnumerable<string> rowKeys)
             return QueryAsync(Query.TimeRange(min, max));
         }
 
-        public async Task<List<ResultVoid<(int, string)>>> BatchOperation(IAsyncEnumerable<T> entities, TableTransactionActionType actionType) {
-            var tableClient = await GetTableClient(typeof(T).Name);
-            var transactions = await entities.Select(e => new TableTransactionAction(actionType, _entityConverter.ToTableEntity(e))).ToListAsync();
-            var responses = await tableClient.SubmitTransactionAsync(transactions);
-            return responses.Value.Select(response =>
-                response.IsError ? ResultVoid<(int, string)>.Error((response.Status, response.ReasonPhrase)) : ResultVoid<(int, string)>.Ok()
-            ).ToList();
+        public async Task<ResultVoid<(int statusCode, string reason, int? failedTransactionIndex)>> BatchOperation(IAsyncEnumerable<T> entities, TableTransactionActionType actionType) {
+            try {
+                var tableClient = await GetTableClient(typeof(T).Name);
+                var transactions = await entities.Select(e => new TableTransactionAction(actionType, _entityConverter.ToTableEntity(e))).ToListAsync();
+                var responses = await tableClient.SubmitTransactionAsync(transactions);
+                var wrappingResponse = responses.GetRawResponse();
+                if (wrappingResponse.IsError) {
+                    return ResultVoid<(int, string, int?)>.Error((wrappingResponse.Status, wrappingResponse.ReasonPhrase, null));
+                }
+
+                var subTransactionFailures = responses.Value.Where(response => response.IsError);
+                if (subTransactionFailures.Any()) {
```

Comment:
isn't this redundant. My understanding is if the `wrappingResponse` is successful then all `subtransactions` are successful. 
I think this logic should be moved inside the check for `wrappingResponse.IsError` to give more detail about what failed in that batch

Reply:
From the docs, it looks like the `wrappingResponse` can be successful while having `subtransactions` that are unsuccessful.

> Assuming that the batch request has been properly authorized and has been successfully received by the Table service, the batch request returns status code 202 (Accepted), even if one of the operations in the change set fails.

https://learn.microsoft.com/en-us/rest/api/storageservices/performing-entity-group-transactions#batch-response-syntax

So there's 3 failure modes that are being handled:

1. The `wrappingResponse` fails and throws a `RequestFailedException`
2. The `wrappingResponse` fails but doesn't throw
3. The `wrappingResponse` succeeds but one of the `subtransactions` fails